homeassistant/auth/providers/legacy_api_password.py

```diff
@@ -3,39 +3,36 @@
 
 It will be removed when auth system production ready
 """
-from collections import OrderedDict
 import hmac
+from collections import OrderedDict
 
 import voluptuous as vol
 
-from homeassistant.exceptions import HomeAssistantError
-from homeassistant import auth, data_entry_flow
+from homeassistant.auth import InvalidAuth
 from homeassistant.core import callback
 
+from . import AUTH_PROVIDER_SCHEMA, AUTH_PROVIDERS, AuthProvider, LoginFlow
+
 USER_SCHEMA = vol.Schema({
     vol.Required('username'): str,
 })
 
 
-CONFIG_SCHEMA = auth.AUTH_PROVIDER_SCHEMA.extend({
+CONFIG_SCHEMA = AUTH_PROVIDER_SCHEMA.extend({
 }, extra=vol.PREVENT_EXTRA)
 
 LEGACY_USER = 'homeassistant'
 
 
-class InvalidAuthError(HomeAssistantError):
```

Comment:
This is an internal error and should stay so. 

Also, changing this is scope creep…